controllers/serviceaccount/serviceaccount_remover_test.go

```diff
@@ -0,0 +1,217 @@
+		doguConfig.Mock.On("Exists", "sa-postgresql").Return(false, assert.AnError)
+		registry := &cesmocks.Registry{}
+		registry.Mock.On("DoguConfig", "redmine").Return(doguConfig)
+		serviceAccountCreator := serviceaccount.NewServiceAccountRemover(registry, nil)
+
+		// when
+		err := serviceAccountCreator.RemoveServiceAccounts(ctx, redmineCr, redmineDescriptor)
+
+		// then
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "failed to check if service account already exists")
+		mock.AssertExpectationsForObjects(t, doguConfig, registry)
+	})
+
+	t.Run("skip routine because serviceaccount does not exist in dogu config", func(t *testing.T) {
+		// given
+		ctx := context.TODO()
+		doguConfig := &cesmocks.ConfigurationContext{}
+		doguConfig.Mock.On("Exists", "sa-postgresql").Return(false, nil)
+		registry := &cesmocks.Registry{}
+		registry.Mock.On("DoguConfig", "redmine").Return(doguConfig)
+		serviceAccountCreator := serviceaccount.NewServiceAccountRemover(registry, nil)
+
+		// when
+		err := serviceAccountCreator.RemoveServiceAccounts(ctx, redmineCr, redmineDescriptor)
+
+		// then
+		require.NoError(t, err)
+		mock.AssertExpectationsForObjects(t, doguConfig, registry)
+	})
+
+	t.Run("failed to check if sa dogu is enabled", func(t *testing.T) {
+		// given
+		ctx := context.TODO()
+		doguConfig := &cesmocks.ConfigurationContext{}
+		doguConfig.Mock.On("Exists", "sa-postgresql").Return(true, nil)
+		doguRegistry := &cesmocks.DoguRegistry{}
+		doguRegistry.Mock.On("IsEnabled", "postgresql").Return(false, assert.AnError)
+		registry := &cesmocks.Registry{}
+		registry.Mock.On("DoguConfig", "redmine").Return(doguConfig)
+		registry.Mock.On("DoguRegistry").Return(doguRegistry)
+		serviceAccountCreator := serviceaccount.NewServiceAccountRemover(registry, nil)
+
+		// when
+		err := serviceAccountCreator.RemoveServiceAccounts(ctx, redmineCr, redmineDescriptor)
+
+		// then
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "failed to check if dogu postgresql is enabled")
+		mock.AssertExpectationsForObjects(t, doguConfig, registry, doguRegistry)
+	})
+
+	t.Run("skip routine because sa dogu is not enabled", func(t *testing.T) {
+		// given
+		ctx := context.TODO()
+		doguConfig := &cesmocks.ConfigurationContext{}
+		doguConfig.Mock.On("Exists", "sa-postgresql").Return(true, nil)
+		doguRegistry := &cesmocks.DoguRegistry{}
+		doguRegistry.Mock.On("IsEnabled", "postgresql").Return(false, nil)
+		registry := &cesmocks.Registry{}
+		registry.Mock.On("DoguConfig", "redmine").Return(doguConfig)
+		registry.Mock.On("DoguRegistry").Return(doguRegistry)
+		serviceAccountCreator := serviceaccount.NewServiceAccountRemover(registry, nil)
+
+		// when
+		err := serviceAccountCreator.RemoveServiceAccounts(ctx, redmineCr, redmineDescriptor)
+
+		// then
+		require.NoError(t, err)
+		mock.AssertExpectationsForObjects(t, doguConfig, registry, doguRegistry)
+	})
+
+	t.Run("failed to get sa descriptor", func(t *testing.T) {
```

Comment:
what is an SA descriptor. Do you mean the consuming dogu object which requests the SA?